src/Storages/ColumnsDescription.cpp

```diff
@@ -659,6 +659,12 @@ bool ColumnsDescription::hasPhysical(const String & column_name) const
         it->default_desc.kind != ColumnDefaultKind::Alias && it->default_desc.kind != ColumnDefaultKind::Ephemeral;
 }
 
+bool ColumnsDescription::hasAlias(const String & column_name) const
+{
+    auto it = columns.get<1>().find(column_name);
+    return it->default_desc.kind == ColumnDefaultKind::Alias;
```

Comment:
Is it guaranteed that the column is found?
Then, maybe we can use `at` instead of `find`?
Otherwise, check for `it`.

Reply:
I see checks for `end()` in the methods above and below.

Reply:
Exactly, that code looks suspicious, but seems like here we have an invariant.

Reply:
Thanks @alexey-milovidov , I will look into this